artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -658,6 +658,8 @@ public QueueImpl(final QueueConfiguration queueConfiguration,
                     final QueueFactory factory) {
       super(server == null ? EmptyCriticalAnalyzer.getInstance() : server.getCriticalAnalyzer(), CRITICAL_PATHS);
 
+      Objects.requireNonNull(server);
```

Comment:
This makes the line above it a bit nonsensical and changing that might make this line  redundant, depending on the approach taken.

Reply:
Agreed. Since Java < 22 forces `super` to be the first call it makes this kind of clumsy. Maybe I should just do:
```
super(server.getCriticalAnalyzer(), CRITICAL_PATHS);
```
And if `server` is `null` then it will throw an NPE instead of `requireNonNull`.

Reply:
Thats probably the option I'd pick this time.